financial-services-accelerator/components/org.wso2.financial.services.accelerator.event.notifications.service/src/main/java/org/wso2/financial/services/accelerator/event/notifications/service/realtime/service/RealtimeEventNotificationSenderService.java

```diff
@@ -0,0 +1,204 @@
+                    }
+                }
+
+                HttpPost httpPost = new HttpPost(URI.create(callbackUrl));
+
+                for (Map.Entry<String, String> entry : httpRequestGenerator.getAdditionalHeaders().entrySet()) {
+                    String headerName = entry.getKey();
+                    String headerValue = entry.getValue();
+                    httpPost.setHeader(headerName, headerValue);
+                }
+
+                httpPost.setEntity(new StringEntity(String.valueOf(payloadJson), ContentType.APPLICATION_JSON));
+                RequestConfig requestConfig = RequestConfig.custom()
+                        .setConnectTimeout(TIMEOUT_IN_SECONDS * 1000)
+                        .setConnectionRequestTimeout(TIMEOUT_IN_SECONDS * 1000)
+                        .setSocketTimeout(TIMEOUT_IN_SECONDS * 1000)
+                        .build();
+                httpPost.setConfig(requestConfig);
+
+                HttpResponse response = httpClient.execute(httpPost);
+                int statusCode = response.getStatusLine().getStatusCode();
+                if (statusCode == HttpStatus.SC_ACCEPTED) {
+                    if (log.isDebugEnabled()) {
+                        log.debug("Real-time event notification with notificationId: " +
+                                notificationId.replaceAll("[\r\n]", "") + " sent successfully");
+                    }
+                    realtimeNotificationService.updateNotificationStatusById(notificationId,
+                            EventNotificationConstants.ACK);
+                    return;
+                } else {
+                    if (log.isDebugEnabled()) {
+                        log.debug("Real-time event notification with notificationId: " +
+                                notificationId.replaceAll("[\r\n]", "")
+                                + " sent failed with status code: " + statusCode);
+                    }
+                }
+
+                // Circuit breaker will be opened if the retrying time exceeds the configured circuit breaker timeout.
+                if (Duration.between(startTime, LocalTime.now()).toMillis()
+                        > CIRCUIT_BREAKER_OPEN_TIMEOUT_IN_SECONDS * 1000) {
+                    circuitBreakerOpen = true;
+                    if (log.isDebugEnabled()) {
+                        log.debug("Circuit breaker open for the realtime event notification with notificationId: "
+                                + notificationId.replaceAll("[\r\n]", ""));
+                    }
+                }
+                retryCount++;
+
+                // If the circuit breaker is opened or the maximum retry count is exceeded,
+                // the notification status will be updated as ERROR.
+                realtimeNotificationService.updateNotificationStatusById(notificationId,
+                        EventNotificationConstants.ERROR);
+            } catch (IOException | InterruptedException  e) {
```

Comment:
if these both exception are logging the same error we can merge them

Reply:
Fixed in https://github.com/wso2/financial-services-accelerator/pull/193/commits/00babc11f6c634ec7023e46cba1a28b73d4377e7